### ISSUE TITLE
Some transactions do not get notes

### ISSUE THREAD

**Issue by jimboca** [alex-peck/monarch-amazon-sync#33]
How to debug a transaction that didn't get any notes?  I have a Monarch Amazon transaction for 11.87 and a matching Amazon order for 11.87 but it's not getting the notes.  Many are getting notes, but also many do not.

Create a donate page!

**Comment by likemondays** [comment]
agreed - I recognize the same issue currently with both the 2.0 and 1.5 extension version, it seems like :(

since September 9th, it looks like 25% or so of transactions get a note, 75% do not :(

**Comment by mattgortner** [comment]
Same here.  The last few weeks, few transactions are getting notes.

This extension is a such a time saver!

**Comment by mattgortner** [comment]
> agreed - I recognize the same issue currently with both the 2.0 and 1.5 extension version, it seems like :(
> 
> since September 9th, it looks like 25% or so of transactions get a note, 75% do not :(
> 
> Edit: just downloaded a 'dry run' .csv from the 0.1.5 extension, and am wondering if there's something wrong with how the Amazon API doesn't return a lot of items in the array [] (see attached), even though otherwise, Monarch and Amazon transactions seem to match up quite okay! [monarch-amazon-matches (3).csv](https://github.com/user-attachments/files/17100448/monarch-amazon-matches.3.csv)

How do you get the CSV?  I downloaded debug logs, but it's a text file.

**Comment by likemondays** [comment]
@mattgortner under manual backfill -> dry run. If that is toggled it spits out a .csv file after the dry run!

@alex-peck any chance y'all had a moment to look into this?
and.. create a donate page.. I think we all feel like we are in your debt - this extension was such a time saver for so many of us!! :-)

**Comment by mattgortner** [comment]
@likemondays  - Thanks - I've tried that a few times, but don't see a link to the file nor see it in downloads.  Where does the file get dumped?

**Comment by likemondays** [comment]
@mattgortner - was going to show you, but as of this morning, I get this error message now

I think the .csv either showed as a download link in the Backfill -> Dry Run dialogue after it completes, or it auto-downloads. One of the two :)

<img width="314" alt="image" src="https://github.com/user-attachments/assets/5311dac9-fa63-4927-85ee-64bd4a44e1c9">

**Comment by likemondays** [comment]
@mattgortner update - switched back to 2.0 from 1.5 and the .csv button is showing up again :)
just in case you were interested!

<img width="368" alt="image" src="https://github.com/user-attachments/assets/99f0303e-d03e-44d4-af16-abe13cad1401">

**Comment by kwc5811** [comment]
My transactions also stopped syncing. I don't see any issues with the extension getting the Amazon orders, debug log confirms that, but when it checks Monarch for Amazon transactions it doesn't find any. Maybe the API for searching for Amazon transactions is broken. I do know that the merchant detail frequently says Amazon.com or Amazon Mkt and then has random letters and numbers.

**Comment by tn1068** [comment]
Same here.  My transactions download from Amazon but getting an error message that it is having trouble fetching Monarch transactions.  My Monarch connection also won't refresh (would be great if there was a way to manually force refresh connections).  I uninstalled and reinstalled the extension, both accounts synced but then when run it didn't find any amazon transactions to match/update

**Comment by tn1068** [comment]
just installed and ran the newest version and all seems to be working well for me again.  Thanks for the updates! The manual account connection refresh is great

**Comment by kwc5811** [comment]
I can also confirm the latest version repaired the syncing! Thank you Alex!

**Comment by likemondays** [comment]
can confirm the same - working again. y'all are legends!! :)

**Comment by mattgortner** [comment]
Awesome, thanks for the fix! Just what we needed after Prime day!

**Comment by kbex9** [comment]
Thank you thank you thank you!

**Comment by shaykb** [comment]
Took me a while to update to this latest release, but it did not resolve the issue for me. In fact, MOST of the transactions are left without comments.
Any ideas in this forum?

**Comment by kwc5811** [comment]
Mine is still working so it must be something specific to your situation. Do you have multiple accounts? Either on the Amazon or Monarch side? Have you tried to force sync or checked the debug logs?

**Comment by shaykb** [comment]
> Mine is still working so it must be something specific to your situation. Do you have multiple accounts? Either on the Amazon or Monarch side? Have you tried to force sync or checked the debug logs?

Thanks a lot for your reply. I do have two separate Amazon accounts that I sync with the same Monarch account. I use separate Chrome profiles for that and did not experience any issues with that in the past (until this problem referenced above appeared around last September). I have forced sync. Have not checked debug logs. Anything specific I should be looking for in those logs?

**Comment by kwc5811** [comment]
Errors? (:

**Comment by shaykb** [comment]
> Errors? (:

Jeez forgot that this is GitHub :-) :-) :-) :-) :-)
Will report back. Thank you...
